app/views/decidim/solutions/admin/solutions/_form.html.erb

```diff
@@ -37,19 +37,59 @@
     </div>
   </div>
 
-  <div class="card-divider">
-    <h2 class="card-title"><%= t(".problem") %></h2>
-  </div>
-
-  <div class="card-section">
-
-    <div class="row column">
-      <%= form.select :decidim_problems_problem_id,
-                      @form.select_problem_collection,
-                      selected: @form.try(:problem) ? [translated_attribute(@form.try(:problem).title), @form.try(:problem).id] : '',
-                      label: t("models.solution.fields.problem", scope: "decidim.solutions.admin"),
-                      include_blank: "" %>
+  <% if current_participatory_space.components.where(manifest_name: "problems").present? %>
+    <div class="card-divider">
+      <h2 class="card-title"><%= t(".problem") %></h2>
+    </div>
+    <div class="card-divider" style="display:none;">
+      <h2 class="card-title"><%= t(".challenge") %></h2>
     </div>
+  <% else %>
+    <div class="card-divider" style="display:none;">
+      <h2 class="card-title"><%= t(".problem") %></h2>
+    </div>
+    <div class="card-divider">
+      <h2 class="card-title"><%= t(".challenge") %></h2>
+    </div>
+  <% end %>
 
-  </div>
-</div>
+  <% if current_participatory_space.components.where(manifest_name: "problems").present? %>
+    <div class="card-section">
+      <div class="row column">
+        <%= form.select :decidim_problems_problem_id,
+                          @form.select_problem_collection,
+                          selected: @form.try(:problem) ? [translated_attribute(@form.try(:problem).title), @form.try(:problem).id] : '',
+                          label: t("models.solution.fields.problem", scope: "decidim.solutions.admin"),
+                          include_blank: "" %>
+      </div>
+    </div>
+    <div class="card-section" style="display:none;">
```

Comment:
why `display: none`? same as before, why just don't render this block?

Reply:
Happens the same in https://github.com/gencat/decidim-challenges/pull/38#discussion_r621045978

Reply:
In that case we can simply render an input type="hidden"